api/lib/opentelemetry/resources/resource.rb

```diff
@@ -9,18 +9,41 @@ module Resources
     # Resource represents a resource, which captures identifying information about the entities
     # for which telemetry (metrics or traces) is reported.
     class Resource
+      # Returns the labels for this {Resource}
+      #
+      # @return [Hash<String, String>]
       attr_reader :labels
 
-      def initialize(**kvs)
+      # Returns a newly created {Resource} with the specified labels
+      #
+      # @param [Hash<String, String>] kvs Hash of key-value pairs to be used
+      #   as labels for this resource
+      # @return [Resource]
+      def initialize(kvs = {})
         # TODO: how defensive should we be here?
```

Comment:
The spec requires the params to be:

> a collection of name/value labels, where name and value are both strings

Should we type-check the keys and values provided?